app/models/assessment.rb

```
@@ -464,6 +464,33 @@ def date_to_s(date)
+  # Public method to access the GDU data
+  def calculate_gdu
+    # Define grade boundaries
+    grade_boundaries = { 'A': 90, 'B': 80, 'C': 70, 'D': 60, 'F': 0 }
+    # Initialize distribution
+    distribution = Hash.new(0)
+
+    total_submissions = submissions.joins(:scores).count
+    submissions.includes(:scores).each do |submission|
+      # Assuming 'score' is a method that sums up scores for a submission
+      score = submission.scores.sum(&:points)
+      grade = grade_boundaries.keys.reverse.detect { |grade| score >= grade_boundaries[grade] }
```

rubocop (1.68.0)

[warning] Shadowing outer local variable - `grade`.

(Lint/ShadowingOuterLocalVariable)